File: balancegame-android/app/src/main/java/com/example/balancegame/api/BalanceService.kt
package com.example.balancegame.api

import com.example.balancegame.model.CatalogGetDto
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface BalanceService {
    //카탈로그 목록을 받아옴
    @GET("/api/1.0/catalogs")
    fun getCatalogsList(
    ): Call<CatalogGetDto>
}

File: balacegame-flutter/lib/api/balanceService.dart
import 'package:dio/dio.dart';
import 'package:balacegame_flutter/model/catalogModel.dart';
import 'package:retrofit/retrofit.dart';
part 'balanceService.g.dart';

@RestApi(baseUrl: 'http://10.0.2.2:9090')
abstract class BalanceService {
  factory BalanceService(Dio dio, {String baseUrl}) = _BalanceService;

  @GET('/api/1.0/catalogs')
  Future<CatalogGetDto> getBalanceGameList();

}